pkgs/by-name/pu/pulumi/with-packages.nix

```diff
@@ -0,0 +1,61 @@
+{
+  lib,
+  runCommand,
+  pulumi,
+  pulumiPackages,
+  makeBinaryWrapper,
+}:
+let
+  version = lib.getVersion pulumi;
+in
+f:
+# Symlink Pulumi plugin executables to the same directory as Pulumi executable.
```

Comment:
Hm, pulumi-yaml expects _resource_ provider schema file in the parent directory of the executable. E.g. `$out/schema-postgresql.json` for `$out/bin/pulumi-resource-postgresql`. So `withPackages` needs a few more tweaks.